### PR TITLE
Add option to use system Trash

### PR DESCRIPTION
**Pull request by xiota**
Add an option to use the system trash instead of immediately deleting files.

screenshot-trash

### CODE REVIEW

File: src/preferencesdialog.ui [hunk 1]
```diff
@@ -117,6 +117,13 @@
          </property>
         </widget>
        </item>
+       <item row="9" column="0" colspan="2">
+        <widget class="QCheckBox" name="useTrashBox">
+         <property name="text">
+          <string>Use system Trash (and don't prompt)</string>
```
Comment (tsujan) [inline src/preferencesdialog.ui:123]: Very good! Please just change "don't" to "do not".

Comment (xiota) [reply src/preferencesdialog.ui]: Done.

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -491,9 +491,14 @@ void MainWindow::on_actionSaveAs_triggered() {
...
 
 void MainWindow::on_actionDelete_triggered() {
-  // delete the current file
-  if(currentFile_)
-    Fm::FileOperation::deleteFiles({currentFile_});
+  // delete or trash the current file
+  if(currentFile_) {
+    if(static_cast<Application*>(qApp)->settings().useTrash()) {
+      Fm::FileOperation::trashFiles({currentFile_}, false);
+    } else {
```
Comment (tsujan) [inline src/mainwindow.cpp:498]: Please also respect the original code style here:

```c++
    }
    else {
```

Comment (xiota) [reply src/mainwindow.cpp]: done.

**Comment by xiota** [comment]
sorry, got branches mixed up... hold on.

**Review by tsujan — APPROVED** [review]
Thanks a lot! GTM.

**Pull request merged**